src/main/java/net/sf/jabref/shared/DBMSSynchronizer.java

```diff
@@ -119,16 +120,19 @@ public void listen(MetaDataChangedEvent event) {
      * Sets the table structure of shared database if needed and pulls all shared entries
      * to the new local database.
      * @param bibDatabase Local {@link BibDatabase}
+     * @throws DatabaseNotSupportedException if the version of shared database does not match
+     *          the version of current shared database support ({@link DBMSProcessor}).
      */
-    public void initializeDatabases() {
-        try {
-            if (!dbmsProcessor.checkBaseIntegrity()) {
-                LOGGER.info(Localization.lang("Integrity check failed. Fixing..."));
-                dbmsProcessor.setUpSharedDatabase();
+    public void initializeDatabases() throws DatabaseNotSupportedException, SQLException {
+        if (!dbmsProcessor.checkBaseIntegrity()) {
+            LOGGER.info(Localization.lang("Integrity check failed. Fixing..."));
+            dbmsProcessor.setUpSharedDatabase();
+
+            if (dbmsProcessor.checkForPre3Dot6Intergrity()) {
```

Comment:
Should that come **before** the `setupSharedDatabase`? I think, the migration warning is displayed at other places, or isn't it? If it is, please add a comment above the if stating the reason why it comes after the setup